aten/src/ATen/native/sparse/cuda/cuSPARSELtOps.cpp

```diff
@@ -0,0 +1,250 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/CUDADataType.h>
+#include <ATen/cuda/CUDASparse.h>
+#include <ATen/cuda/CUDAConfig.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/Dispatch.h>
+#include <c10/core/ScalarType.h>
+#include <c10/cuda/CUDACachingAllocator.h>
+#include <c10/util/CallOnce.h>
+#include <c10/util/Half.h>
+#include <cusparse.h>
+#include <cstdint>
+#include <Exceptions.h>
+
+
+#if AT_CUSPARSELT_ENABLED()
```

Comment:
nit would be simpler to have a single if/else for this I think.
At the top, when it is not defined, define the API that error. Then the rest of the file is the else where it is always available.